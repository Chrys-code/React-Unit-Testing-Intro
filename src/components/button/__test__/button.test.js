import "@testing-library/jest-dom/extend-expect";
import React from "react";
import ReactDOM from "react-dom";
import Button from "../button";
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";

// Functionality
// Renders & unmounts correctly
it("renders whithout crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Button />, div);
  ReactDOM.unmountComponentAtNode(div);
});

// Style test
// Rendered regarding the X properties
it("renders button correctly", () => {
  const { getByTestId } = render(<Button label="Click Me!" />);
  expect(getByTestId("button")).toHaveTextContent("Click Me!");
});

it("renders button correctly", () => {
  const { getByTestId } = render(<Button label="Click!" />);
  expect(getByTestId("button")).toHaveTextContent("Click!");
});

// Snapshots
// First run
// Create a snapshot file & put the element into it as JSON
// Why? - Pushing to version control, your testing will handle snapshots as well.
// More runs
// Compare snapshots to previous snapshots in the snapshot file.
// Decide if the change within the React element was intentionally and necessary.

it("matches snapshot 1", () => {
  const tree = renderer.create(<Button label="Click!" />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("matches snapshot 2", () => {
  const tree = renderer.create(<Button label="Click Me!" />).toJSON();
  expect(tree).toMatchSnapshot();
});
