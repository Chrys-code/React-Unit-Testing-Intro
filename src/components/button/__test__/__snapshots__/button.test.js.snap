// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1 1`] = `
<button
  className="button-style"
  data-testid="button"
>
  Hi,
  Click!
</button>
`;

exports[`matches snapshot 2 1`] = `
<button
  className="button-style"
  data-testid="button"
>
  Hi,
  Click Me!
</button>
`;
